core/controllers/profile.py

```diff
@@ -370,31 +370,40 @@ def put(self):
 
 
 class UserInfoHandler(base.BaseHandler):
-    """Provides info about user."""
+    """Provides info about user. If user is not logged in,
+    return dict containing false as logged in status.
+    """
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
-    @acl_decorators.require_user_id
+    @acl_decorators.open_access
     def get(self):
```

Comment:
> There should be backend tests covering all the different conditional cases.

I am serious about testing _all_ the cases. There are three branches (username exists, user_id exists but not username, user is not logged in) so I would expect to see three tests. This is core functionality, we need to make sure that it doesn't cause regressions.

Reply:
Done

Reply:
A user can't be created without a username

Reply:
Then why not just do "if self.username"?

Reply:
Yeah
I didn't even think of that.
Okay I'll do that.